lib/src/v3/protocol.dart

```diff
@@ -32,63 +33,71 @@ class AggregatedClientMessage extends ClientMessage {
   }
 }
 
-StreamChannelTransformer<BaseMessage, List<int>> messageTransformer =
-    StreamChannelTransformer(
-  _readMessages,
-  StreamSinkTransformer.fromHandlers(
-    handleData: (message, out) {
-      if (message is! ClientMessage) {
-        out.addError(
-            ArgumentError.value(message, 'message', 'Must be a client message'),
-            StackTrace.current);
-        return;
-      }
-
-      out.add(message.asBytes());
-    },
-  ),
-);
-
-StreamTransformer<Uint8List, ServerMessage> _readMessages =
-    StreamTransformer.fromBind((rawStream) {
-  return Stream.multi((listener) {
-    final framer = MessageFramer();
-
-    var paused = false;
-
-    void emitFinishedMessages() {
-      while (framer.hasMessage) {
-        listener.addSync(framer.popMessage());
-
-        if (paused) break;
-      }
-    }
-
-    void handleChunk(Uint8List bytes) {
-      framer.addBytes(bytes);
-      emitFinishedMessages();
-    }
-
-    final rawSubscription = rawStream.listen(handleChunk)
-      ..onError(listener.addErrorSync);
-
-    listener.onPause = () {
-      paused = true;
-      rawSubscription.pause();
-    };
-
-    listener.onResume = () {
-      paused = false;
-      emitFinishedMessages();
-
-      if (!paused) {
-        rawSubscription.resume();
-      }
-    };
-
-    listener.onCancel = () {
-      paused = true;
-      rawSubscription.cancel();
-    };
-  });
-});
+class WrapMessageTransformer {
```

Comment:
Turning this into a class with `late` fields seems unnecessary, especially since `_readMessages` is only used to construct the public `messageTransformer`.

Maybe we can change the definitions of `_readMesages` and `messageTransformer` to be top-level methods (outside of any class) that take the encoding?

```dart
StreamChannelTransformer<BaseMessage, List<int>> messageTransformer(Encoding encoding) {
  return StreamChannelTransformer(
    _readMessages(encoding),
   ...
  );
}

StreamTransformer<Uint8List, ServerMessage> _readMessages(Encoding encoding) {
  ...
}
```

Reply:
fixed